Xplat/src/main/java/vazkii/botania/common/block/flower/generating/EntropinnyumBlockEntity.java

```diff
@@ -29,54 +27,77 @@
 import vazkii.botania.common.handler.BotaniaSounds;
 import vazkii.botania.xplat.XplatAbstractions;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class EntropinnyumBlockEntity extends GeneratingFlowerBlockEntity {
 	private static final int RANGE = 12;
 	private static final int EXPLODE_EFFECT_EVENT = 0;
 	private static final int ANGRY_EFFECT_EVENT = 1;
 
+	private static boolean trackTntEntities = false;
+	private static final List<PrimedTnt> trackedTntEntities = new ArrayList<>();
+
 	public EntropinnyumBlockEntity(BlockPos pos, BlockState state) {
 		super(BotaniaFlowerBlocks.ENTROPINNYUM, pos, state);
 	}
 
-	public static boolean isUnethical(Entity e) {
-		BlockPos center = e.blockPosition();
-		if (!e.level.isLoaded(center)) {
-			return false;
+	/**
+	 * A force relay, force lens mana burst or vanilla piston is about to start moving blocks.
+	 */
+	public static void startTrackingTntEntities() {
+		assert (!trackTntEntities && trackedTntEntities.isEmpty());
+		trackTntEntities = true;
```

Comment:
Similarly, should this be a counter (in case some crazy mixin makes moveBlocks reentrant)?

Reply:
> 

Yes, if the tnt block is activated after updating, a tnt entity will be generated when the tnt block is pushed. For the case of piston driven tnt, this method should be feasible. When the tnt is copied, it will become a 36 block. Perhaps you can simplify the detection idea from here?